File: frontend/src/types/department.ts
import { DayOfWeek } from './index';

export type DepartmentFormValues = {
    name: string;
    description?: string;
    color?: string;
    startTime?: string;
    endTime?: string;
    workingDays?: DayOfWeek[];
    defaultSchedulePattern?: string;
};

export type Department = {
    id: number
    name: string
    description?: string
    color: string
    startTime?: string;
    endTime?: string;
    workingDays: DayOfWeek[];
    defaultSchedulePattern: SchedulePattern;
    autorenewSchedules: boolean
}

export enum SchedulePattern {
    Flexible = 0,
    TwoOnTwoOff = 1,
    FiveOnTwoOff = 2,
    ThreeOnThreeOff = 3,
    FourOnFourOff = 4
}

//Mapper
export const SchedulePatternOptions = [
    {
        label: "Flexible",
        value: "Flexible",
        enumValue: SchedulePattern.Flexible,
    },
    {
        label: "2 on / 2 off",
        value: "TwoOnTwoOff",
        enumValue: SchedulePattern.TwoOnTwoOff,
    },
    {
        label: "5 on / 2 off",
        value: "FiveOnTwoOff",
        enumValue: SchedulePattern.FiveOnTwoOff,
    },
]

File: backend/Domain/Department.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Enums;

namespace Domain;

public class Department
{
    [Key]
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Color { get; set; } = "#3b82f6";
    public bool AutorenewSchedules { get; set; } = false;
    public TimeSpan? StartTime { get; set; }
    public TimeSpan? EndTime { get; set; }
    public List<DayOfWeek> WorkingDays { get; set; } = new();
    public SchedulePattern DefaultSchedulePattern { get; set; } = SchedulePattern.Flexible;
        
    [ForeignKey("Organization")]
    public int OrganizationId { get; set; }
    public Organization? Organization { get; set; }

    public ICollection<EmployeeInDepartment> EmployeeDepartments { get; set; } = new List<EmployeeInDepartment>();
    public ICollection<ShiftTemplate> ShiftTemplates { get; set; } = new List<ShiftTemplate>();
}